liver_imaging_analysis/engine/engine.py

```diff
@@ -394,22 +395,38 @@ def fit(
             # reset the status for next computation round
             self.metrics.reset()
             # every evaluate_epochs, test model on test set
-            if (epoch + 1) % evaluate_epochs == 0:  
-                valid_loss, valid_metric = self.test(self.test_dataloader)
+            if (epoch + 1) % evaluate_epochs == 0: 
+                if (epoch + 1) % metric_epoch == 0:  
+                    valid_loss, valid_metric = self.test(self.test_dataloader, post_process=True)
+                else:
+                    valid_loss = self.test(self.test_dataloader, post_process=False) 
             if save_weight:
                 self.save_checkpoint(save_path)
             else:
                 valid_loss = None
                 valid_metric = None
-            self.per_epoch_callback(
+                
+            if (epoch + 1) % metric_epoch == 0:
+                self.per_epoch_callback(
+                        epoch,
+                        training_loss,
+                        valid_loss,
+                        training_metric,
+                        valid_metric,
+                        metric_epoch,
+                    )
+            else:
+                self.per_epoch_callback(
                     epoch,
                     training_loss,
                     valid_loss,
                     training_metric,
-                    valid_metric,
+                    None,
```

Comment:
why would you need to do this?

Reply:
because this error : local variable 'valid_metric' referenced before assignment